src/utils.rs

```diff
@@ -732,6 +833,112 @@ fn char_width(c: char) -> usize {
     }
 }
 
+/// Returns the number of visible characters, ignoring style related chars
+fn count_visible_chars(text: &str) -> usize {
+    let mut total_visible = 0;
+
+    let mut chars = text.chars();
+    while let Some(c) = chars.next() {
+        if c == '\x1b' {
+            while let Some(next) = chars.next() {
+                if next == 'm' {
+                    break;
+                }
+            }
+            continue;
+        }
+        if c != '\n' {
+            total_visible += 1;
+        }
+    }
+
+    total_visible
+}
+
+/// Applies the gradient over a string
+fn apply_gradient_impl(
+    text: &str,
+    gradients: &Vec<ColorGradient>,
```

Comment:
Should take a `&[_]` instead of `&Vec<_>`.

Reply:
Indeed, corrected